adobe_analytics_2/aanalytics2.py

```diff
@@ -909,6 +909,13 @@ def getReport(
         page_nb += 1
         if verbose:
             print(f"# of requests : {page_nb}")
+    # Filter out error Columns
+    if "columnErrors" in report["columns"].keys():
+        for column_error in report["columns"]["columnErrors"]:
+            column_name = next((column["id"] for column in json_request["metricContainer"]["metrics"] if column["columnId"] == column_error["columnId"]), None)
```

Comment:
I wonder, whether a list comprehension is the best choice here (instead of list + for loop + if condition).
Wasn't there a max characters per line recommendation?

Reply:
I must admit the list comprehension was the result of a quick "stack-overflow" on how to easily search for a dictionary item fullfilling a specific condition.

Maybe just using filter or reduce with a lambda is more readable.
What do you think?

Reply:
As Mathematician, I'm all for Functional Programming.
However, Python code doesn't read so nice with it …

```py
column_id = None
for column in json_request["metricContainer"]["metrics"]:
    if column["columnId"] == column_error["columnId"]:
        column_id = column["id"]

column_name = next(column_id, None)
```
is more Pythonic, even if it takes more lines.

Reply:
```python
column_name = None
for column in json_request["metricContainer"]["metrics"]:
    if str(column["columnId"]) == str(column_error["columnId"]):
        column_name = column["id"]
        break
```
might be the best alternative then. 
I'll leave that up to @pitchmuc to decide.

Reply:
Following Andre recommendation, I will modify the code to reflect a more pythonic way to solve this.
Final code will be : 
```
    if "columnErrors" in report["columns"].keys():
        for column_error in report["columns"]["columnErrors"]:
            column_name = None
            for column in json_request["metricContainer"]["metrics"]:
                if str(column["columnId"]) == str(column_error["columnId"]):
                    column_name = column["id"]
                    if verbose:
                        print(f"Warning : not able to get column {column_name}: {column_error['errorDescription']}")
                    break
            columns.remove(column_name)
```